src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -327,6 +327,9 @@ private function convertParameters(array $parameters, string $type, \DOMElement
                 $element->setAttribute('type', 'abstract');
                 $text = $this->document->createTextNode(self::phpToXml($value->getText()));
                 $element->appendChild($text);
+            } elseif ($value instanceof \UnitEnum) {
```

Comment:
Shouldn't this `if` include PHP version check? `\PHP_VERSION_ID >= 80100`?

Reply:
Recently, I also made #40830 to add support of PHP 8.1 enumerations for the serializer component. What comes out is we should not do version check at this kind of place: if someone wants to do a `UnitEnum` polyfill for PHP < 8.1, we should allow it.

[Here is the direct link to the thread about it.](https://github.com/symfony/symfony/pull/40830#discussion_r614699836)

Reply:
Also, there's nothing to safe-guard here: The `if` can safely be executed on PHP 7 and if we ever raise the minimum PHP version to 8.1 or above, we would still need to perform the same check.

Let's not add a PHP version check here.